### PR TITLE
fix: remove --allow-run warning when using deno without args or subcommand

### PR DESCRIPTION
**Pull request by HasanAlrimawi**
This PR addresses issue #25676 

**Approach**
Don't warn when no args deno is used.   _[command typed as `deno`]_

**Changes**
- A check was added before warning about the allow-run. 
- Given that _no_args_deno_ is dealt with as a `deno repl`. The check is to see if the subcommand is a `repl `command with the is_default_command property true, if yes, then this is a _no_args_deno_ and there's no need to warn.
- If it's not `no_args_deno` and the `allow-run` flag is found, then the warning is shown.

**Issue root-cause**
when using no_args_deno (invoking deno with no args or subcommands), then all of the `allow-*` flags are set by default which caused the warning to get triggered because there's no allow-run-list specified with it.

### CODE REVIEW

File: cli/args/mod.rs [hunk 1]
```diff
@@ -1702,8 +1702,13 @@ fn warn_insecure_allow_run_flags(flags: &Flags) {
...
 
-  // discourage using --allow-run without an allow list
-  if allow_run_list.is_empty() {
+  let is_no_args_deno = match flags.subcommand.clone() {
+    DenoSubcommand::Repl(repl_flags) => repl_flags.is_default_command,
+    _ => false,
+  };
+  // discourage using --allow-run without an allow list, but
+  // allow it when no args deno is used [deno] without subcommand
+  if allow_run_list.is_empty() && !is_no_args_deno {
```
Comment (dsherret) [inline cli/args/mod.rs:1711]: I think we should change this to instead set `--allow-all`: https://github.com/denoland/deno/blob/f360cae9dd1150c1ca02d51631ddaddb5bf871ab/cli/args/flags.rs#L1370-L1376

**Review by dsherret — CHANGES_REQUESTED** [review]

File: cli/args/mod.rs (hunk 1 above)
Comment (dsherret) [reply cli/args/mod.rs:1711]: There's actually a `flags.allow_all();` and I think internally that should just set `flags.permissions.allow_all = true` only.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks!

**Comment by dsherret** [comment]
I broke this. Looking into it.

**Comment by HasanAlrimawi** [comment]
I tried to investigate it:
Mainly, I believe whenever allow-all is set, it should allow access for everything except if there's specific deny flag.
Maybe the issue lies within permission granting logic, I suspect the fix might be somewhere in cli/runtime/permissions/lib.rs, as your changes seem logical (allow all has high priority and there's no need to **set** other allow flags to `Some(vec![])` and the logic should be good with them cleared as `None`)

I noticed you made a new commit to fix the issue, but let me ask you to take a look at `fn is_allow_all()` in cli/runtime/permissions/lib.rs. Does it need to check if the allow-all flag is set to true at some point and if yes then does it do that.
The function name seems convey this idea.

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**